File: Java/RemoveDuplicatesfromSortedArray.java
public class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length == 0){
            return 0;
        }

        int currentLength = 1;
        
        for(int i = 1; i < nums.length; ++i){
            if(nums[i] != nums[i-1]){
                nums[currentLength] = nums[i];
                ++currentLength;
            }
        }
        return currentLength;
    }
}

File: C++/RemoveDuplicatesfromSortedArray.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int length = nums.size();
        if(length == 0)
            return 0;
        
        int count {1};
        for(int i = 1; i < length; ++i){
            if(nums[i-1] != nums[i])
                nums[count++] = nums[i];
        }
        return count;
    }
};
